src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php

```diff
@@ -431,12 +438,16 @@ public function iWantToModifyAProduct(ProductInterface $product): void
         $this->sharedStorage->set('product', $product);
 
         if ($product->isSimple()) {
-            $this->updateSimpleProductPage->open(['id' => $product->getId()]);
+            $this->testHelper->waitUntilPageOpens(3, function() use ($product): void {
+                $this->updateSimpleProductPage->open(['id' => $product->getId()]);
+            });
```

Comment:
If we have such a specific method, what do you think about passing the page instance and route options instead of the callable?

```suggestion
            $this->testHelper->waitUntilPageOpens(3, $this->updateSimpleProductPage, ['id' => $product->getId()]);
```

And we could even define a default timeout and use it as the last argument so that it could be even shorter:

```suggestion
            $this->testHelper->waitUntilPageOpens($this->updateSimpleProductPage, ['id' => $product->getId()]);
```